lib/onelogin/ruby-saml/response.rb

```
@@ -528,14 +528,40 @@ def validate_audience
       def validate_destination
         return true if destination.nil? || destination.empty? || settings.assertion_consumer_service_url.nil? || settings.assertion_consumer_service_url.empty?
 
-        unless destination == settings.assertion_consumer_service_url
+        unless uri_match?
           error_msg = "The response was received at #{destination} instead of #{settings.assertion_consumer_service_url}"
           return append_error(error_msg)
         end
 
         true
       end
 
+      # Compare the destination and the ACS url.  The scheme and the FQDN should be matched case insensitive, while
+      # the path should be case sensitive. If Rails can't parse out a scheme and a host, default to the
+      # original match.  If a URI
+      # @return [Boolean]
+      def uri_match?
+        dest_uri = URI.parse(destination)
+        acs_uri = URI.parse(settings.assertion_consumer_service_url)
+
+        if dest_uri.scheme.nil? || acs_uri.scheme.nil? || dest_uri.host.nil? || acs_uri.host.nil?
+          original_uri_match?
```

Maybe this should just raise `URI::InvalidURIError` and count on the rescue block to take care of it?

Otherwise seems reasonable to me.  What's the plan if OneLogin doesn't accept this PR?  Did we confirm how many companies were affected by this?  Was it just SNEI?